### ISSUE TITLE
ChatGLM2 Pretrain 后合并模型推理死循环

### ISSUE THREAD

**Issue by Peter-L-FANG** [shibing624/MedicalGPT#168]
### Describe the bug
Please provide a clear and concise description of what the bug is. If applicable, add screenshots to help explain your problem, especially for visualization related problems.
使用中文语料数据进行pretrain后，合并模型推理循环输出，无法停止

训练使用单机单卡：
CUDA_VISIBLE_DEVICES=0 torchrun --nproc_per_node 1 pretraining.py \
    --model_type chatglm \
    --model_name_or_path ./chatglm2-6b \
    --train_file_dir ./data/pretrain \
    --validation_file_dir ./data/pretrain \
    --per_device_train_batch_size 2 \
    --per_device_eval_batch_size 2 \
    --do_train \
    --do_eval \
    --use_peft True \
    --seed 42 \
    --fp16 \
    --max_train_samples 10000 \
    --max_eval_samples 10 \
    --num_train_epochs 1 \
    --learning_rate 2e-4 \
    --warmup_ratio 0.05 \
    --weight_decay 0.01 \
    --logging_strategy steps \
    --logging_steps 10 \
    --eval_steps 50 \
    --evaluation_strategy steps \
    --save_steps 500 \
    --save_strategy steps \
    --save_total_limit 3 \
    --gradient_accumulation_steps 1 \
    --preprocessing_num_workers 1 \
    --block_size 1024 \
    --output_dir outputs-pt-v1 \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --target_modules all \
    --lora_rank 8 \
    --lora_alpha 16 \
    --lora_dropout 0.05 \
    --torch_dtype float16 \
    --device_map auto \
    --report_to tensorboard \
    --ddp_find_unused_parameters False \
    --gradient_checkpointing True

合并脚本：
python merge_peft_adapter.py --model_type chatglm \
            --base_model_name_or_path chatglm2-6b --peft_model_path outputs-pt-v1 --output_dir merged-pt/

![image](https://github.com/shibing624/MedicalGPT/assets/65850906/c43d3b88-01ee-42c1-ad89-9eaf5f607f36)

**Comment by shibing624** [comment]
PT阶段和SFT阶段不同，对话是在SFT后才具备的功能

**Comment by sevenandseven** [comment]
> ### Describe the bug
> Please provide a clear and concise description of what the bug is. If applicable, add screenshots to help explain your problem, especially for visualization related problems. 使用中文语料数据进行pretrain后，合并模型推理循环输出，无法停止
> 
> 训练使用单机单卡： CUDA_VISIBLE_DEVICES=0 torchrun --nproc_per_node 1 pretraining.py --model_type chatglm --model_name_or_path ./chatglm2-6b --train_file_dir ./data/pretrain --validation_file_dir ./data/pretrain --per_device_train_batch_size 2 --per_device_eval_batch_size 2 --do_train --do_eval --use_peft True --seed 42 --fp16 --max_train_samples 10000 --max_eval_samples 10 --num_train_epochs 1 --learning_rate 2e-4 --warmup_ratio 0.05 --weight_decay 0.01 --logging_strategy steps --logging_steps 10 --eval_steps 50 --evaluation_strategy steps --save_steps 500 --save_strategy steps --save_total_limit 3 --gradient_accumulation_steps 1 --preprocessing_num_workers 1 --block_size 1024 --output_dir outputs-pt-v1 --overwrite_output_dir --ddp_timeout 30000 --logging_first_step True --target_modules all --lora_rank 8 --lora_alpha 16 --lora_dropout 0.05 --torch_dtype float16 --device_map auto --report_to tensorboard --ddp_find_unused_parameters False --gradient_checkpointing True
> 
> 合并脚本： python merge_peft_adapter.py --model_type chatglm --base_model_name_or_path chatglm2-6b --peft_model_path outputs-pt-v1 --output_dir merged-pt/
> 
> ![image](https://private-user-images.githubusercontent.com/65850906/260997736-c43d3b88-01ee-42c1-ad89-9eaf5f607f36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA5ODY0NDksIm5iZiI6MTcxMDk4NjE0OSwicGF0aCI6Ii82NTg1MDkwNi8yNjA5OTc3MzYtYzQzZDNiODgtMDFlZS00MmMxLWFkODktOWVhZjVmNjA3ZjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIxVDAxNTU0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjNDQ5MWE4YmZmNWE2YjM0M2Q5Zjg1NWI5Nzc2MmVjOTViNGFkYWFhMTBiMWY2MzljOGNiNWVkNWRjZWY4YTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HBZ5XbEGnSswYIM2qBXtKLdOscTeqNpix7K6J398Gy4)

请问：合并模型推理循环输出，无法停止，这个问题怎么解决？

你的训练效果看起来比较好，大概有多少条数据、训练了多少代，有做过数据清洗吗？
